alica_engine/src/engine/teammanager/Agent.cpp

```diff
@@ -67,14 +65,15 @@ bool Agent::update()
     if (_local) {
         return false;
     }
-    if (_active && _timeLastMsgReceived + _timeout < _clock.now()) {
+    const auto time = _timeLastMsgReceived.load() + _timeout;
+    if (_active && time < _clock.now()) {
```

Comment:
Assuming the clock is monotonic `now - timelast > timeout` can be safer for addition overflow.

Reply:
it's a  `AlicaTime` class. Hope it could handle a overflow. And initial state is empty, so using `-` could be more strange due underflow problem

Reply:
Maybe not empty, but default...